platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -560,4 +594,11 @@ public void debugDumpTransactionCache() {
     private int computeBucketIndex(final int keyHash) {
         return (numOfBuckets - 1) & keyHash;
     }
+
+    private record ReadBucketResult<K extends VirtualKey>(Bucket<K> bucket, Throwable error) {
+        public ReadBucketResult {
+            assert (bucket != null) || (error != null);
```

Comment:
assert isn't usually enabled at runtime; should these be if tests that throw error if failed?
Also, this is perhaps better expressed as an XOR, rather than two OR conditions, particularly as I believe these two will throw an AssertionError in all cases (if either or both values are null, then if either or both values are not null).
I assume what you meant to write is `assert (bucket == null) ^ (error == null)` so one must be null, and the other not null, but you cannot have both null or both not null.
Possibly better approach:
```java
if(!((bucket == null) ^ (error == null))) {
  throw error == null ? error : new AssertionError("ReadBucketResult cannot have both bucket and error, but must have exactly one of those values");
}
```

Reply:
XOR is a good idea, I forgot Java has `^` operator as it's so rarely used ) There is no need to enable this assertion in runtime, this record usages are very limited and straightforward.